### ISSUE TITLE
An error occurs when clicking the bin button on the Data Retention window

### ISSUE THREAD

**Issue by tvanzeleDelen** [trimble-oss/dba-dash#410]
After making some changes to the data retention, I wanted to run it immediately. I clicked the bin button, but this gives an error:
![image](https://user-images.githubusercontent.com/101806473/197710977-96a09cff-2106-4c0c-b3db-a0b15689a15f.png)

**Comment by DavidWiseman** [comment]
Hi,

Thank you for reporting this - I'll get this fixed in the next release.  If you want to apply a hotfix, you can run this:

```sql
DROP TABLE [Switch].[JobStats_60MIN] 
GO
CREATE TABLE [Switch].[JobStats_60MIN] (
    [InstanceID]        INT              NOT NULL,
    [job_id]            UNIQUEIDENTIFIER NOT NULL,
    [step_id]           INT              NOT NULL,
    [RunDateTime]       DATETIME2 (2)    NOT NULL,
    [FailedCount]       INT              NOT NULL,
    [SucceededCount]    INT              NOT NULL,
    [RetryCount]        INT              NOT NULL,
    [RunDurationSec]    INT              NOT NULL,
    [MaxRunDurationSec] INT              NOT NULL,
    [MinRunDurationSec] INT              NOT NULL,
    CONSTRAINT [PK_JobStats_60MIN] PRIMARY KEY CLUSTERED ([InstanceID] ASC, [job_id] ASC, [step_id] ASC, [RunDateTime] ASC)
) 
```

The "Switch.JobStats_60MIN" doesn't have any data.  The old partitions are switched to this table to efficiently remove the old data.  The problem is the Switch.JobStats_60MIN is also partitioned so a partition number would need to be specified.  The fix is to remove the `ON PS_JobStats_60MIN(RunDateTime)` from the table definition so the partition switching works the same as the other tables.

Another fix you could apply is to increase the retention of the JobStats_60MIN table so the cleanup doesn't run for this table.  This will prevent the error and allow the cleanup to run for the other tables.  The JobStats_60MIN table is only 27MB - it just stores the aggregated stats so it's cheaper to store the data long term.

Thanks,

David

PS

The app calls dbo.PurgeData when you click the bin icon.  This will then call dbo.PartitionTable_Cleanup for each table with the specified retention.  This could also be called manually to remove data for a specific table. e.g.

`EXEC dbo.PartitionTable_Cleanup @TableName='SessionWaits',@DaysToKeep=7`

**Comment by DavidWiseman** [comment]
Fixed in 2.24.2 🎉
